src/content/resources/java/streams.md

```diff
@@ -0,0 +1,75 @@
+---
+authors:
+  - "KaeseKuchenDEV#6322"
+title: "An introduction to Streams"
+created_at: 2019/12/15
+---
+
+# An introduction to Streams
+Basically everyone, who spent a few hours coding some Java came across a situation like this:
+
+```java
+int[] intArray = {1,2,3,4,5};
+
+for(int i = 0; i < intArray.length; i++) {
+  System.out.println(intArray[i]);
+}
+```
+
+So, you need to print every element in an array, a list or any other data structure.
+But there has to be a nicer way to do this, especially with less code. Since Java 8 we are
+able to use so called "Streams" in Java. In particular the same result as from the code above
+can be achieved with this code:
+
+```java
+int[] intArray = {1,2,3,4,5};
+
+Arrays.stream(intArray).forEach(n -> System.out.println(n));
+```
+
+You just made one line out of three. But how does this work?
+`int[] intArray = {1,2,3,4,5};` does just declare and initialize a new array of integers. This stays the
+same as before, because you can use the `Arrays.stream()` method for every Array,
+[as it works with generic Arrays](https://www.mkyong.com/java8/java-how-to-convert-array-to-stream/). This method creates a [Stream Object](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html),
+that has several methods to perform operations on an array.
+
+##Sorting
+
+Now maybe you have a list instead of an array and you also want the list to get sorted before you
+print it. Without Streams you would first have to sort the list and then iterate through it.
+Way too many lines! Using Streams you can simply do:
+
+```java
+ArrayList<Integer> intList = new ArrayList<>();
+intList.add(4);
+intList.add(8);
+intList.add(3);
+intList.add(6);
+
+intList.stream().sorted().forEach(n -> System.out.print(n + ", "));
+```
+
+At the end, you have again only one line, for sorting and printing every element in a list.
+Due to the fact that Streams mostly use Method Chaining, you could perform a bunch of operations.
+
+##Filtering
```

Comment:
While we're at it, maybe mention .map as well?

Reply:
Yeah, i just created this PR to explain a bit of the basics of streams and add some other things later